packages/ubuntu_wsl_setup/lib/services/named_event.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'dart:collection';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+/// Exposes a [future] that completes when a platform event named [name] is set.
+class NamedEvent {
+  /// The event name as known by the platform, UTF-8 encoded.
+  String name;
+  final _completer = Completer<void>();
+
+  /// Completes when the platform event is set.
+  Future<void> get future => _completer.future;
+  NamedEvent(this.name) {
```

Comment:
- could be const after making name final
- add empty line before the constructor

Reply:
brainfart, cannot be const with a body :D

Reply:
Still the String name being final is useful. 😅